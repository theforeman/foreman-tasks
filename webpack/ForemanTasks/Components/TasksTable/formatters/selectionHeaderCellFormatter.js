import React from 'react';
import TableSelectionHeaderCell from '../Components/TableSelectionHeaderCell';

export default (selectionController, label) => (
  <TableSelectionHeaderCell
    label={label}
    checked={selectionController.allPageSelected()}
    disabled={!selectionController.permissions.edit}
    onChange={selectionController.selectPage}
  />
);
