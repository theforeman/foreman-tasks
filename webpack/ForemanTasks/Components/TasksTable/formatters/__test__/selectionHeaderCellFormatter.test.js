import selectionHeaderCellFormatter from '../selectionHeaderCellFormatter';

describe('selectionHeaderCellFormatter', () => {
  it('render', () => {
    expect(
      selectionHeaderCellFormatter(
        { allPageSelected: () => true, permissions: { edit: true } },
        'some-label'
      )
    ).toMatchSnapshot();
  });
});
