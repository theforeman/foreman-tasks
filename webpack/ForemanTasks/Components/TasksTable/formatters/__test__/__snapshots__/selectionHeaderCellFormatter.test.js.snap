// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectionHeaderCellFormatter render 1`] = `
<TableSelectionHeaderCell
  checked={true}
  disabled={false}
  id="selectAll"
  label="some-label"
  onChange={[Function]}
/>
`;
